tests/lang/ArrayTest.php

```diff
@@ -560,4 +560,30 @@ public function testInsertAssociativeArray(): void {
 		$this->assertEquals(4, $alimony->size());
 		$this->assertEquals(['potatoes', 'carrots'], $alimony->get(3));
 	}
+
+	public function testAddNotOverwriteElements(): void {
+		$fruits = new ArrayObject(['apple', 'banana', 'peach']);
+		$this->assertEquals(3, $fruits->size());
+
+		$fruits->remove('banana');
+		$this->assertEquals(2, $fruits->size());
+		$this->assertEquals(['apple', 'peach'], $fruits->toArray());
+		$fruits->add('apricot');
+
+		$this->assertEquals(3, $fruits->size());
+		$this->assertEquals(['apple', 'peach', 'apricot'], $fruits->toArray());
```

Comment:
you maybe wanna add:

```php
$this->assertTrue(array_is_list($fruits->toArray());
```

in some places? insert/add/remove on _all_ lists?

Reply:
Good catch! Done.